src/start_server.ts

```
@@ -71,7 +76,8 @@ export function startServer(options: ServerOptions): Promise<http.Server> {
         resolve(options);
       });
     }
-  }).then<http.Server>((opts) => startWithPort(opts));
+  }).then<http.Server>((opts) => startWithPort(opts))
+    .catch((e) => console.error('ERROR: Server failed to start:', e));
```

You need to propagate the error, so throw it after the `console.error`